include/albatross/src/models/sparse_gp.hpp

```diff
@@ -531,22 +517,103 @@ class SparseGaussianProcessRegression
     //   = y^T y_a - y_b^T y_b
     //
     // with y_b = R^-T K_uf y_a
-    const Eigen::VectorXd y_a = sc.A_ldlt.solve(sc.y);
+    const Eigen::VectorXd y_a = A_ldlt.solve(y);
 
-    Eigen::VectorXd y_b = sc.K_fu.transpose() * y_a;
-    y_b = sqrt_solve(sc.B_qr, y_b);
+    Eigen::VectorXd y_b = K_fu.transpose() * y_a;
+    y_b = sqrt_solve(B_qr, y_b);
 
-    double log_quadratic = sc.y.transpose() * y_a;
+    double log_quadratic = y.transpose() * y_a;
     log_quadratic -= y_b.transpose() * y_b;
 
-    const double rank = static_cast<double>(sc.y.size());
+    const double rank = static_cast<double>(y.size());
     const double log_dimension = rank * log(2 * M_PI);
 
     return -0.5 * (log_det + log_quadratic + log_dimension) +
            this->prior_log_likelihood();
   }
 
 private:
+  // This method takes care of a lot of the common book keeping required to
+  // setup the Sparse Gaussian Process problem.  Namely, we want to get from
+  // possibly unordered features to a structured representation
+  // in the form K_ff = A_ff + Q_ff where Q_ff = K_fu K_uu^-1 K_uf and
+  // A_ff is block diagonal and is formed by subtracting Q_ff from K_ff.
+  //
+  template <typename InducingFeatureType, typename FeatureType>
+  void compute_internal_components(
```

Comment:
Both uses of this function are immediately followed with
` const auto B_qr = compute_sigma_qr(K_uu_ldlt, A_ldlt, K_fu);`
Does it make sense to combine `compute_sigma_qr` and `compute_internal_components` in some way?

Reply:
There's actually a third use in `_update_impl` which doesn't follow it with `B_qr` which is why that's split out.